app/containers/ChatContainer/reducer.js

```diff
@@ -0,0 +1,30 @@
+/*
+ *
+ * ChatContainer reducer
+ *
+ */
+
+import { fromJS } from 'immutable';
+import {
+  SEND_MESSAGE_ACTION,
+  RECEIVE_MESSAGE_ACTION,
+} from './constants';
+
+const initialState = fromJS({
+  messages: [],
+});
+
+function chatContainerReducer(state = initialState, { type, payload }) {
+  switch (type) {
+    case SEND_MESSAGE_ACTION:
+      return state;
+    case RECEIVE_MESSAGE_ACTION: {
+      const messages = state.get('messages');
```

Comment:
Gotcha here is that our state is immutable, so we need to make sure and use immutable methods to update it.